pkg/utils/validator/clusterresourceplacement_test.go

```diff
@@ -1036,3 +1038,122 @@ func TestIsPlacementPolicyUpdateValid(t *testing.T) {
+		},
+		"invalid toleration, value is invalid, operator is Equal": {
+			tolerations: []placementv1beta1.Toleration{
+				{
+					Key:      "key1",
+					Operator: corev1.TolerationOpEqual,
+					Value:    "val#%/-123",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+			wantErr: fmt.Errorf(invalidTolerationValueErrFmt, placementv1beta1.Toleration{Key: "key1", Operator: corev1.TolerationOpEqual, Value: "val#%/-123", Effect: corev1.TaintEffectNoSchedule}, "a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')"),
+		},
+		"invalid toleration, key is invalid, operator is Exists": {
+			tolerations: []placementv1beta1.Toleration{
+				{
+					Key:      "key:123*",
+					Operator: corev1.TolerationOpExists,
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+			wantErr: fmt.Errorf(invalidTolerationKeyErrFmt, placementv1beta1.Toleration{Key: "key:123*", Operator: corev1.TolerationOpExists, Effect: corev1.TaintEffectNoSchedule}, "name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')"),
+		},
+		"invalid toleration, value is not empty, operator is Exists": {
+			tolerations: []placementv1beta1.Toleration{
+				{
+					Key:      "key1",
+					Operator: corev1.TolerationOpExists,
+					Value:    "value1",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+			wantErr: fmt.Errorf(invalidTolerationErrFmt, placementv1beta1.Toleration{Key: "key1", Operator: corev1.TolerationOpExists, Value: "value1", Effect: corev1.TaintEffectNoSchedule}, "toleration value needs to be empty, when operator is Exists"),
+		},
+		"invalid toleration, non-unique toleration": {
+			tolerations: []placementv1beta1.Toleration{
+				{
+					Key:      "key1",
+					Operator: corev1.TolerationOpEqual,
+					Value:    "value1",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+				{
+					Key:      "key2",
+					Operator: corev1.TolerationOpExists,
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+				{
+					Key:      "key1",
+					Operator: corev1.TolerationOpEqual,
+					Value:    "value1",
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			},
+			wantErr: fmt.Errorf(uniqueTolerationErrFmt, placementv1beta1.Toleration{Key: "key1", Operator: corev1.TolerationOpEqual, Value: "value1", Effect: corev1.TaintEffectNoSchedule}),
+		},
+	}
+	for testName, testCase := range tests {
+		t.Run(testName, func(t *testing.T) {
+			if actualErr := validateTolerations(testCase.tolerations); actualErr != nil && actualErr.Error() != testCase.wantErr.Error() {
```

Comment:
```suggestion
			if got := validateTolerations(testCase.tolerations); actualErr != nil && actualErr.Error() != testCase.wantErr.Error() {
```

prefer to use "got" & "want" instead of "actual" & "expected"

" Where you would write “actual” and “expected”, prefer using the words “got” and “want”, respectively."

https://google.github.io/styleguide/go/decisions#got-before-want